include/deal.II/lac/sparse_matrix.templates.h

```diff
@@ -1898,6 +1898,80 @@ SparseMatrix<number>::get_sparsity_pattern () const
 }
 
 
+namespace internal
+{
+  namespace SparseMatrix
+  {
+    template <class StreamType, typename realType>
+    inline
+    void print_value(StreamType &out, const realType&val)
+    {
+      out << val;
+    }
+
+    template<class StreamType, typename realType>
+    inline
+    void print_value(StreamType &out, const std::complex<realType> &val)
+    {
+      // the following format is what is required by the matrix market format.
+      out << val.real << ' ' << val.imag;
+    }
+
+    template<class StreamType, class numberType>
+    void
+    print_header(StreamType &out,const numberType & dummy)
+    {
```

Comment:
Thank you!  I was about to give up.  I was going to ask for a better way to do that (without the dummy variable).  A few questions if you have a minute.

1. Does this idea / pattern have a name?
2. Can you explain the use of static?
3. Am I correct in assuming you meant `print_header` for the "member function" instead of `print_value`?




Can you explain the use of static?